project/src/types/user-info-type.ts

```diff
@@ -0,0 +1,15 @@
+export type AbstractUser = {
+  id?: number,
+  email?: string,
+  name?: string,
+  token?: string;
+}
+
+export type UserInfoType = null | AbstractUser & {
+  avatarUrl?: string,
+};
+
+export type ServerUserInfoType = AbstractUser & {
+  'avatar_url': string,
```

Comment:
Тут 
```suggestion
  avatar_url: string,
```
не сработает?